src/final2/subtests/InvalidCommandTest.java

```diff
@@ -0,0 +1,915 @@
+			};
+			sessionTest(runs, Input.getFile(inputFile));
+
+			inputFile = new String[] {
+					"0000",
+					"0000",
+					"0a00",
+					"0000"
+			};
+			runs = new Run[] {
+					new ErrorRun("create h," + invalidNumber + ",1"),
+					quit()
+			};
+			sessionTest(runs, Input.getFile(inputFile));
+
+			inputFile = new String[] {
+					"0000",
+					"0000",
+					"0a00",
+					"0000"
+			};
+			runs = new Run[] {
+					new ErrorRun("field 1," + invalidNumber),
+					quit()
+			};
+			sessionTest(runs, Input.getFile(inputFile));
+
+			inputFile = new String[] {
+					"0000",
+					"0000",
+					"0a00",
+					"0000"
+			};
+			runs = new Run[] {
+					new ErrorRun("field " + invalidNumber + ",1"),
+					quit()
+			};
+			sessionTest(runs, Input.getFile(inputFile));
+		}
+	}
+
+	/**
+	 * Asserts that the program reacts correctly on commands with missing arguments
+	 */
+	@Test
+	public void missingArgumentTest() {
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("move"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"b000",
+				"0000",
+				"0a00",
+				"Z000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create h,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"b000",
+				"0000",
+				"0a00",
+				"Z000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create 2,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"b000",
+				"0000",
+				"0a00",
+				"Z000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"b000",
+				"0000",
+				"0a00",
+				"Z000"
+		};
+		runs = new Run[] {
+				new ErrorRun("escape"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("position"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("field 1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("field"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("direction"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("escape"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+	}

```

Comment:
I'd somewhere (not necessarily in this ^ method) add tests that test e.g. <code>create&nbsp;</code> (i.e. with just a trailing space) etc. Some people might do `cmd.split(" ")` and then use e.g. `.charAt(0)` on the second element of the split array.